bike_rent/models/sale.py

```diff
@@ -11,17 +11,23 @@ def action_confirm(self):
         # <AndriusD code
         def filter_rests(rec):
             product = rec.product_id
-            return product.is_bike and product.type == 'service'
-
-        for record in self.order_line.filtered(filter_rests):  # /AndriusD code>
-            rent_rec = self.env['bike.rent'].create({
-                'notes': record.name,
-                'price': record.price_total,
-                'partner_id': record.order_partner_id.id,
-                'bike_id': record.product_id.id,
-                'rent_time': record.product_id.rent_duration,
-            })
-            record.bike_rent_id = rent_rec.id
+            return product.is_bike and product.type == 'service'  # /AndriusD code>
+
+        def create_bike_rent_objects(so_record):
```

Comment:
`so_record` is not `sale.order`, it's `sale.order.line` so `line` or `so_line` would be more appropriate. Also could be method declared in model,  as so:
```
Class SaleOrderLine(models.Model):
    _inherit = 'sale.order.line'

     def create_bike_rents(self):
        for line in self:
```